compiler/rustc_mir/src/borrow_check/diagnostics/mutability_errors.rs

```diff
@@ -502,6 +515,84 @@ impl<'a, 'tcx> MirBorrowckCtxt<'a, 'tcx> {
 
         err.buffer(&mut self.errors_buffer);
     }
+    
+    /// User cannot make signature of a trait mutable without changing the
+    /// trait. So we find if this error belongs to a trait and if so we move
+    /// suggestion to the trait or disable it if it is out of scope of this crate                            
+    fn is_error_in_trait(&self, local: Local) -> (bool, Option<Span>) {
+        if self.body.local_kind(local) != LocalKind::Arg {
+            return (false, None);
+        }
+        let hir_map = self.infcx.tcx.hir();
+        let my_hir = hir_map.local_def_id_to_hir_id(
+            self.body.source.def_id().as_local().unwrap(),
+        );
+        match hir_map.find(hir_map.get_parent_node(my_hir)) {
```

Comment:
This long match can be replaced by `tcx.opt_associated_item` to get an `AssocItem`, and then `tcx.trait_id_of_impl` to get the `DefId` of the trait.

Reply:
I replaced long match with `tcx.impl_of_method`.